### ISSUE TITLE
MongoDB's `nestIn` and `nestNotIn` are impossible to use and should be removed

### ISSUE THREAD

**Issue by TomMD** [yesodweb/persistent#1083]
nestNotIn (`$nin`) takes an array of values and filters out any record where the field is an element of this array.

In the persistent library though, `nestNotIn` does not take a collection such as a list or set and instead takes a single element - this is just nonsense.

**Comment by parsonsmatt** [comment]
@TomMD You're the only person I know of that actually uses `persistent-mongoDB`. Would you like to be a maintainer for it?

**Comment by TomMD** [comment]
@parsonsmatt No, I'm trying to shed maintainer-ships not gain them.  Thank you, though, both for this and general Haskell community efforts.

I realize we need some significant effort in both the mongodb driver and persistent integration (not to dismiss all the work - it's a lot to build and then a lot more could be built).  It isn't yet clear how to proceed with Haskell + NoSQL.  The resources available are quite limited but a rich MongoDB library is both a significant cost and a significant benefit so I'll keep mulling things.  The first thing is probably to shore up the underlying driver to allow TLS, support rich and easy group operations, and support the new wire format.  Only then can we as a community address the persistent overlay and receive significant real-world benefit.

**Comment by parsonsmatt** [comment]
Totally fair :) Best of luck!
